src/plugins/intel_npu/src/plugin/npuw/partitioning/partitioning.cpp

```diff
@@ -1693,9 +1714,17 @@ void Partitioner::optimize(const std::string& func_name) {
 
             ov::parallel_for(func_group.refs.size(), [&](std::size_t f_idx) {
                 auto& funcall = func_group.refs[f_idx].get();
-                ov::Tensor cw = funcall._closure[w_idx - f._param_offset];
-                ov::Tensor cz = z_idx != -1 ? funcall._closure[z_idx - f._param_offset] : ov::Tensor{};
-                ov::Tensor cs = funcall._closure[s_idx - f._param_offset];
+                // FIXME: assuming no transformations were applied to the tensor - since we are utilizing the original
+                // ov::Tensor below
+                NPUW_ASSERT(!funcall._lazy_closure[w_idx - f._param_offset].has_transformations());
+                if (z_idx != -1) {
+                    NPUW_ASSERT(!funcall._lazy_closure[z_idx - f._param_offset].has_transformations());
+                }
+                NPUW_ASSERT(!funcall._lazy_closure[s_idx - f._param_offset].has_transformations());
+                ov::Tensor cw = funcall._lazy_closure[w_idx - f._param_offset].get_orig_tensor();
+                ov::Tensor cz =
+                    z_idx != -1 ? funcall._lazy_closure[z_idx - f._param_offset].get_orig_tensor() : ov::Tensor{};
+                ov::Tensor cs = funcall._lazy_closure[s_idx - f._param_offset].get_orig_tensor();
```

Comment:
How the three lines of code before turned into this?
It should be the same three lines of code. 

Also, why do you work with `ov::Tensor`s here??

Reply:
1) Will remove asserts
2) What do you mean? It's the same as was before

Reply:
1. Done

Reply:
2. Are you suggesting to operate with `LazyTensor` here?